cohort-engine/src/main/java/com/ibm/cohort/engine/measure/seed/MeasureEvaluationSeeder.java

```diff
@@ -0,0 +1,179 @@
+package com.ibm.cohort.engine.measure.seed;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.tuple.Triple;
+import org.hl7.fhir.r4.model.Library;
+import org.hl7.fhir.r4.model.Measure;
+import org.opencds.cqf.common.evaluation.EvaluationProviderFactory;
+import org.opencds.cqf.common.helpers.DateHelper;
+import org.opencds.cqf.common.helpers.UsingHelper;
+import org.opencds.cqf.common.providers.LibraryResolutionProvider;
+import org.opencds.cqf.cql.engine.data.DataProvider;
+import org.opencds.cqf.cql.engine.debug.DebugMap;
+import org.opencds.cqf.cql.engine.execution.Context;
+import org.opencds.cqf.cql.engine.execution.LibraryLoader;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.terminology.TerminologyProvider;
+import org.opencds.cqf.r4.helpers.LibraryHelper;
+
+import com.google.common.annotations.VisibleForTesting;
+
+public class MeasureEvaluationSeeder {
+
+	private final EvaluationProviderFactory providerFactory;
+	private final LibraryLoader libraryLoader;
+	private final LibraryResolutionProvider<Library> libraryResourceProvider;
+
+	private boolean disableExpressionCaching;
+	private boolean debugMode = true;
+	private String terminologyProviderSource;
+	private String terminologyProviderUser;
+	private String terminologyProviderPassword;
+
+	public MeasureEvaluationSeeder(
+			EvaluationProviderFactory providerFactory,
+			LibraryLoader libraryLoader,
+			LibraryResolutionProvider<Library> libraryResourceProvider) {
+		this.providerFactory = providerFactory;
+		this.libraryLoader = libraryLoader;
+		this.libraryResourceProvider = libraryResourceProvider;
+	}
+
+	public MeasureEvaluationSeeder disableDebugLogging() {
+		this.debugMode = false;
+
+		return this;
+	}
+
+	public MeasureEvaluationSeeder disableExpressionCaching() {
+		this.disableExpressionCaching = true;
+
+		return this;
+	}
+
+
+	public MeasureEvaluationSeeder withTerminologyProvider(String source, String user, String password) {
+		this.terminologyProviderSource = source;
+		this.terminologyProviderUser = user;
+		this.terminologyProviderPassword = password;
+
+		return this;
+	}
+
+
+	public IMeasureEvaluationSeed create(Measure measure, String periodStart, String periodEnd, String productLine) {
+		LibraryHelper.loadLibraries(measure, this.libraryLoader, this.libraryResourceProvider);
+
+		// resolve primary library
+		org.cqframework.cql.elm.execution.Library library =
+                LibraryHelper.resolvePrimaryLibrary(measure, libraryLoader, this.libraryResourceProvider);
+
+		List<Triple<String, String, String>> usingDefs = UsingHelper.getUsingUrlAndVersion(library.getUsings());
```

Comment:
It might be worth adding a comment somewhere to help fresh people understand what `Triple<String, String, String>` actually means in this context.

Reply:
this is actually code I had to copy from the `cql-engine` code.
So I'm not even sure I fully understand the purpose.

https://github.com/Alvearie/quality-measure-and-cohort-service/pull/5#discussion_r512787332 has the conversation on it for more context.